### PR TITLE
Make WMCO run on hostNetwork

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
As of now, WMCO runs on pod network. This PR ensures that WMCO runs on
hostNetwork. This is a step towards making WMCO use machine api.
The advantage of running WMCO on hostNetwork includes not having a public IP
for the Windows VM created. This relies on
openshift/windows-machine-config-bootstrapper#214

The master branch of WNI is having issues while setting up SSH client. So pinning WNI in go.mod to
https://github.com/openshift/windows-machine-config-bootstrapper/commit/727c39ac62ac0b7622029f3ffc0eccbbb44e5693

### CODE REVIEW

**Comment by ravisantoshgudimetla** [comment]
I tested this locally and it works with a 4.5 cluster:

```2020-06-22T22:05:28.695Z	DEBUG	controller_wmc	updating status	{"status": {"joinedVMCount":0,"conditions":[{"type":"Reconciling","status":"True","lastTransitionTime":"2020-06-22T22:05:28Z"}]}}
2020-06-22T22:05:28.704Z	DEBUG	controller_wmc	creating a Windows VM
2020/06/22 22:05:29 Using existing Security Group: sg-014c20f4dfa0fa9f3
2020/06/22 22:09:32 External IP: 
2020/06/22 22:09:32 Internal IP: 10.0.165.127
2020-06-22T22:09:32.486Z	DEBUG	controller_wmc	configuring the Windows VM	{"ID": "i-00e835de04d43698c"}
2020-06-22T22:09:47.208Z	DEBUG	windows	ignition file download	{"cmd": "C:\\Temp\\wget-ignore-cert.ps1 -server https://api-int.ravig120.devcluster.openshift.com:22623/config/worker -output C:\\Windows\\Temp\\worker.ign", "output": ""}
2020-06-22T22:09:50.303Z	DEBUG	windows	output from wmcb	{"output": "Bootstrapping completed successfully"}
```

File: pkg/controller/windowsmachineconfig/tracker.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 	nc "github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachineconfig/nodeconfig"
 	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
 	"github.com/pkg/errors"
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (ravisantoshgudimetla) [inline pkg/controller/windowsmachineconfig/tracker.go:14]: Note to myself: Remove `v1` alias

File: deploy/namespace.yaml [hunk 2]
```diff
@@ -2,3 +2,5 @@ apiVersion: v1
 kind: Namespace
 metadata:
   name: windows-machine-config-operator
+  labels:
+    openshift.io/run-level: "1"
```
Comment (sebsoto) [inline deploy/namespace.yaml:6]: > I can give you a label selector to use temporarily that is not something you can GA with that will skip SCC. any namespace with a label "openshift.io/run-level = 1" will skip SCC.  But you still have a general problem with providing a way to safely validate pods that want to run on windows and SCC must not be skipped on user requests if the pod is a linux pod.

This is from our discussion of the Windows webhook. This does not seem to be the correct way to go about this

Comment (ravisantoshgudimetla) [reply deploy/namespace.yaml:6]: > providing a way to safely validate pods that want to run on windows and SCC must not be skipped on **user** requests if the pod is a linux pod.

My understanding of the above statement is - it is correct for user request pods not the cluster ones. For the Windows specific pods that run in any namespace we shouldn't add this label or ask user to add this label for the namespace.

Comment (ravisantoshgudimetla) [reply deploy/namespace.yaml:6]: To be clear, only `windows-machine-config-operator` namespace should have this label and the WMCO pod should be the only pod running in this namespace.

Comment (sebsoto) [reply deploy/namespace.yaml:6]: I am not too familar with the terminology, so I did not focus on the `user requests` part. That makes things better.

I am still hesitant about this solution. Instead of removing just the `hostnetwork` blocking you are removing all SCC. It seems like we should be more focused on what we are removing. I am worried about vulnerabilities that could arise from this.

**Comment by mansikulkarni96** [comment]
LGTM

File: deploy/namespace.yaml (hunk 2 above)
Comment (ravisantoshgudimetla) [reply deploy/namespace.yaml:6]: I'll see if I can associate WMCO service account with hostNetwork SCC and see if it works.

Comment (aravindhp) [reply deploy/namespace.yaml:6]: For my understanding, you are using the label just so that the pod can run on the `host-network`. If so, what do other components that need to run on the `host-network` do?

File: pkg/controller/windowsmachineconfig/tracker.go (hunk 1 above)
Comment (aravindhp) [reply pkg/controller/windowsmachineconfig/tracker.go:14]: PS: I really would like to figure out why this keeps getting added and stop that from happening.

File: go.mod [hunk 3]
```diff
@@ -10,6 +10,7 @@ replace (
 	github.com/mattn/go-sqlite3 => github.com/mattn/go-sqlite3 v1.10.0
 	github.com/openshift/api => github.com/openshift/api v0.0.0-20200422081840-fdd1b0c14c88 // OpenShift 4.5
 	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20200422192633-6f6c07fc2a70 // OpenShift 4.5
+	github.com/openshift/windows-machine-config-bootstrapper/tools/windows-node-installer => github.com/openshift/windows-machine-config-bootstrapper/tools/windows-node-installer v0.0.0-20200619143322-727c39ac62ac
```
Comment (aravindhp) [inline go.mod:13]: Why is this needed?

File: go.mod [hunk 4]
```diff
@@ -36,7 +37,7 @@ replace (
 require (
 	github.com/openshift/api v0.0.0-20200422081840-fdd1b0c14c88
 	github.com/openshift/client-go v0.0.0-20200422192633-6f6c07fc2a70
-	github.com/openshift/windows-machine-config-bootstrapper/tools/windows-node-installer v0.0.0-20200619015040-0288e17404c0
+	github.com/openshift/windows-machine-config-bootstrapper/tools/windows-node-installer v0.0.0-20200620102919-dd78396c3ed4
```
Comment (aravindhp) [inline go.mod:40]: FYI: This hash includes https://github.com/openshift/windows-machine-config-bootstrapper/pull/215. I thought you did not want to include that. Just calling that out. I have no problems with you doing so.

File: deploy/namespace.yaml (hunk 2 above)
Comment (ravisantoshgudimetla) [reply deploy/namespace.yaml]: Atleast MCO uses this label on the namespace they need.

File: go.mod (hunk 3 above)
Comment (ravisantoshgudimetla) [reply go.mod:13]: To overwrite what we have in require

File: go.mod (hunk 4 above)
Comment (ravisantoshgudimetla) [reply go.mod:40]: https://github.com/openshift/windows-machine-config-operator/pull/72/files#r444469531

File: pkg/controller/windowsmachineconfig/tracker.go (hunk 1 above)
Comment (ravisantoshgudimetla) [reply pkg/controller/windowsmachineconfig/tracker.go]: It should come from go-imports but in vscode, I changed to gofmt but I see this happening. IDK why either :(

**Comment by ravisantoshgudimetla** [comment]
Tested this locally on 4.5 cluster and it is working fine:

```2020/06/23 19:44:48 Internal IP: 10.0.151.48
2020-06-23T19:44:48.617Z	DEBUG	controller_wmc	configuring the Windows VM	{"ID": "i-03d997237e34f874e"}
2020-06-23T19:45:17.562Z	DEBUG	windows	ignition file download	{"cmd": "C:\\Temp\\wget-ignore-cert.ps1 -server https://api-int.ravig121.devcluster.openshift.com:22623/config/worker -output C:\\Windows\\Temp\\worker.ign", "output": ""}
2020-06-23T19:45:24.408Z	DEBUG	windows	output from wmcb	{"output": "Bootstrapping completed successfully"}
```

File: deploy/olm-catalog/windows-machine-config-operator/manifests/windows-machine-config-operator.clusterserviceversion.yaml [hunk 5]
```diff
@@ -191,6 +192,15 @@ spec:
           - '*'
           verbs:
           - '*'
+        # Permissions to access hostnetwork SCC
```
Comment (sebsoto) [inline deploy/olm-catalog/windows-machine-config-operator/manifests/windows-machine-config-operator.clusterserviceversion.yaml:195]: Is this generated or did you manually add this?
The file should be generated from the deploy yamls

Comment (ravisantoshgudimetla) [reply deploy/olm-catalog/windows-machine-config-operator/manifests/windows-machine-config-operator.clusterserviceversion.yaml:195]: Ahh sorry, I should've called it. The generation as of now is broken atleast for me. As of now, if you do `operator-sdk generate csv`, I am noticing that all the permissions are being taken out. I am using v0.18.1 of operator-sdk

`operator-sdk version: "v0.18.1", commit: "7bf7b6886d647dc202525daec16fab67dcc52a3d", kubernetes version: "v1.18.2", go version: "go1.13.6 linux/amd64"`

Comment (aravindhp) [reply deploy/olm-catalog/windows-machine-config-operator/manifests/windows-machine-config-operator.clusterserviceversion.yaml:195]: Please bring this up with the operator SDK team and call it out in the README.

Comment (sebsoto) [reply deploy/olm-catalog/windows-machine-config-operator/manifests/windows-machine-config-operator.clusterserviceversion.yaml:195]: if the generation doesnt work because of your changes thats a problem, if its unavoidable we need to change the readme to remove the generation

Comment (PratikMahajan) [reply deploy/olm-catalog/windows-machine-config-operator/manifests/windows-machine-config-operator.clusterserviceversion.yaml:195]: I dont understand what you mean by `permissions are being take out` ?

File: go.mod (hunk 3 above)
Comment (aravindhp) [reply go.mod:13]: Can't you `go get` to a particular commit hash? That way you won't need to do this override.

File: go.mod (hunk 4 above)
Comment (aravindhp) [reply go.mod:40]: Can't you `go get` to a particular commit hash?

File: deploy/olm-catalog/windows-machine-config-operator/manifests/windows-machine-config-operator.clusterserviceversion.yaml (hunk 5 above)
Comment (PratikMahajan) [reply deploy/olm-catalog/windows-machine-config-operator/manifests/windows-machine-config-operator.clusterserviceversion.yaml:195]: i tried running the command on my machine. I was able to generate the CSV with required permissions. Not sure why this is happening on your machine. 
i am on 
```
operator-sdk version: "v0.18.1", commit: "7bf7b6886d647dc202525daec16fab67dcc52a3d", kubernetes version: "v1.18.2", go version: "go1.13.6 linux/amd64"
```

File: deploy/olm-catalog/windows-machine-config-operator/manifests/windows-machine-config-operator.clusterserviceversion.yaml [hunk 6]
```diff
@@ -171,13 +178,6 @@ spec:
           - deployments/finalizers
           verbs:
           - update
-        - apiGroups:
-            - ""
```
Comment (ravisantoshgudimetla) [inline deploy/olm-catalog/windows-machine-config-operator/manifests/windows-machine-config-operator.clusterserviceversion.yaml:175]: @PratikMahajan - this is what I was talking about

**Comment by aravindhp** [comment]
/lgtm cancel

Sorry I did not mean to LGTM but mean to cancel it. @PratikMahajan given you have an open discussion, please lgtm once that is concluded.

File: deploy/olm-catalog/windows-machine-config-operator/manifests/windows-machine-config-operator.clusterserviceversion.yaml (hunk 6 above)
Comment (PratikMahajan) [reply deploy/olm-catalog/windows-machine-config-operator/manifests/windows-machine-config-operator.clusterserviceversion.yaml:175]: just saw that they're just being rearranged.
